src/vos/vos_layout.h

```diff
@@ -246,12 +246,15 @@ enum vos_io_stream {
 	VOS_IOS_CNT
 };
 
+#define DAOS_CONT_V1	0x0001
+#define DAOS_CONT_VERSION	DAOS_CONT_V1
 /* VOS Container Value */
 struct vos_cont_df {
 	uuid_t				cd_id;
 	uint64_t			cd_nobjs;
 	uint32_t			cd_ts_idx;
-	uint32_t			cd_pad;
+	uint16_t			cd_ver;
```

Comment:
We already maintain the POOL_DF version in pd_version field.   This is updated on pool upgrade.  Do we need to support more than one container version at a time?  Can we just iterate containers in the pool in vos_pool_upgrade and reallocate each container root so as to maintain a single container version per pool?

Reply:
The idea here would be to control this using pool->vp_feats which is setup on initialization based on the pool_df.

Reply:
sure, will update the patch. Thanks.